pkg/byteorder/byteorder.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2022-2023 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package byteorder
+
+import (
+	"encoding/binary"
+	"unsafe"
+)
+
+// GetHostByteOrder returns the endianness of the CPU.
+func GetHostByteOrder() binary.ByteOrder {
```

Comment:
nite: we only support little endian hosts, don't think the perf hit is too high, but this little-endian assumption is backed in various places in the agent

Reply:
Happy to remove it in subsequent PRs